manifests/base/application-controller/argocd-application-controller-statefulset.yaml

```diff
@@ -57,3 +60,15 @@ spec:
                 matchLabels:
                   app.kubernetes.io/part-of: argocd
               topologyKey: kubernetes.io/hostname
+      volumes:
+      - name: argocd-repo-server-tls
+        secret:
+          secretName: argocd-repo-server-tls
```

Comment:
Just had a discussion about a simiar requirement for Redis - we will need to store tls and password for Redis as well.

To avoid creating multiple secrets can we remove `-tls` from secret name and just have secret per component: `argocd-repo-server`, `argocd-redis`?

Reply:
I'm not very opinionated about the actual name of the Secret. But I think it would be good to have Secrets specific to TLS, which does not contain other configuration, because people might have external controllers to manage this kind of information (e.g. Cert Manager, or some other TLS manager to distribute valid certs and keys). I also think that we might want to migrate the `tls.crt` and `tls.key` for the API server out of the `argocd-secret` to its own secret in the future.

My proposal would be to set the Redis password in `argocd-secret` secret, and the `tls.crt` and `tls.key` data in a dedicated secret.